pkg/repo/migrations/v3_4.go

```diff
@@ -0,0 +1,28 @@
+package migrations
+
+import (
+	"os"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/repo"
+)
+
+// V3Migration deletes the network store (NATS KV store) so that it may recreated once the node starts.
+// It deletes the store since in v1.3.0 the store contains models.NodeInfo and in v1.3.1 the store
+// contains models.NodeState and these types are incompatible with each other.
+// Full details may be found in https://github.com/bacalhau-project/bacalhau/issues/4024
+var V3Migration = repo.NewMigration(
+	repo.RepoVersion3,
+	repo.RepoVersion4,
+	func(r repo.FsRepo) error {
+		repoPath, err := r.Path()
+		if err != nil {
+			return err
+		}
+		resolvedCfg, err := config.Load(repoPath)
+		if err != nil {
+			return err
+		}
+		return os.RemoveAll(resolvedCfg.Node.Network.StoreDir)
```

Comment:
Actually `Node.Network.StoreDir` gives the parent
`~/.bacalhau/orchestrator_store` which holds both job store and nats store. I don't recall the motivation for doing it this way.

I am also concerned that we delete something not intended if the user misconfigured this path to be something outside of the bacalhau dir and something we shouldn't delete.

If we were to go with the repo migration route instead of migrating entries, then we need to validate this path   , and only delete specific sub-directories that are related to node kv store and not the whole path

Reply:
> Actually Node.Network.StoreDir gives the parent
~/.bacalhau/orchestrator_store which holds both job store and nats store. I don't recall the motivation for doing it this way.

I don't think that is the case see https://github.com/bacalhau-project/bacalhau/blob/main/pkg/config/config.go#L87 and attached debugger screenshot.
![image](https://github.com/bacalhau-project/bacalhau/assets/6546409/958d7ebd-f4ab-49e7-bac1-5cff2e72f236)

We can check if the path is outside the repo, and if that is the case abort the migration, but I think the odds of that happening are next to zero.